src/recyclable_block_flyout_inflater.js

```diff
@@ -0,0 +1,164 @@
+      const xml =
+        typeof blockDefinition["blockxml"] === "string"
+          ? Blockly.utils.xml.textToDom(blockDefinition["blockxml"])
+          : blockDefinition["blockxml"];
+      return xml.getAttribute("type");
+    } else {
+      return blockDefinition["type"];
+    }
+  }
+
+  /**
+   * Puts a previously created block into the recycle bin and moves it to the
+   * top of the workspace. Used during large workspace swaps to limit the number
+   * of new DOM elements we need to create.
+   *
+   * @param block The block to recycle.
+   */
+  recycleBlock(block) {
+    const xy = block.getRelativeToSurfaceXY();
+    block.moveBy(-xy.x, -xy.y);
+    this.recycledBlocks.set(block.type, block);
+  }
+
+  /**
+   * Returns a block from the cache of recycled blocks with the given type, or
+   * undefined if one cannot be found.
+   *
+   * @param blockType The type of the block to try to recycle.
+   * @returns The recycled block, or undefined if
+   *     one could not be recycled.
+   */
+  getRecycledBlock(blockType) {
+    const block = this.recycledBlocks.get(blockType);
+    this.recycledBlocks.delete(blockType);
+    return block;
+  }
+
+  /**
+   * Returns whether the given block can be recycled or not.
+   *
+   * @param block The block to check for recyclability.
+   * @returns True if the block can be recycled. False otherwise.
+   */
+  blockIsRecyclable(block) {
+    if (!this.recyclingEnabled) {
+      return false;
+    }
+
+    // If the block needs to parse mutations, never recycle.
+    if (block.mutationToDom && block.domToMutation) {
+      return false;
+    }
+
+    if (!block.isEnabled()) {
+      return false;
+    }
+
+    for (const input of block.inputList) {
+      for (const field of input.fieldRow) {
+        // No variables.
+        if (field.referencesVariables()) {
+          return false;
+        }
+        if (field instanceof Blockly.FieldDropdown) {
+          if (field.isOptionListDynamic()) {
+            return false;
+          }
+        }
+      }
+      // Check children.
+      if (input.connection) {
+        const targetBlock =
+          /** @type {Blockly.BlockSvg} */
+          (input.connection.targetBlock());
+        if (targetBlock && !this.blockIsRecyclable(targetBlock)) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Disposes of the provided block.
+   *
+   * @param {!Blockly.BlockSvg} element The block to dispose of.
+   */
+  disposeElement(element) {
+    if (this.blockIsRecyclable(element)) {
+      this.removeListeners(element.id);
+      this.recycleBlock(element);
+    } else {
+      super.disposeElement(element);
+    }
+  }
+
+  /**
+   * Clears the cache of recycled blocks.
+   */
+  emptyRecycledBlocks() {
+    this.recycledBlocks
+      .values()
+      .forEach((block) => block.dispose(false, false));
+    this.recycledBlocks.clear();
+  }
+}
+
+/**
+ * Registers the recyclable block flyout inflater.
```

Comment:
Add "replacing the standard block flyout inflator".

Reply:
Done